File: server/src/Logger.js
class Logger {
  constructor() {
    this.logs = [];
    this.maxLogs = 1000;
  }

  log(level, message, data = null) {
    const entry = {
      timestamp: new Date().toISOString(),
      level,
      message,
      data
    };
    
    this.logs.push(entry);
    
    if (this.logs.length > this.maxLogs) {
      this.logs.shift();
    }
    
    const formatted = `[${level.toUpperCase()}] ${message}${data ? ' ' + JSON.stringify(data) : ''}`;
    console.log(formatted);
    
    return entry;
  }

  info(message, data) {
    return this.log('info', message, data);
  }

  error(message, data) {
    return this.log('error', message, data);
  }

  warn(message, data) {
    return this.log('warn', message, data);
  }

  debug(message, data) {
    return this.log('debug', message, data);
  }

  getRecentLogs(count = 100) {
    return this.logs.slice(-count);
  }

  clear() {
    this.logs = [];
  }
}

module.exports = new Logger();

File: app/lib/services/logger.dart
class Logger {
  static void info(String message, [dynamic data]) {
    print('[INFO] $message${data != null ? ' $data' : ''}');
  }

  static void error(String message, [dynamic data]) {
    print('[ERROR] $message${data != null ? ' $data' : ''}');
  }

  static void debug(String message, [dynamic data]) {
    print('[DEBUG] $message${data != null ? ' $data' : ''}');
  }

  static void warn(String message, [dynamic data]) {
    print('[WARN] $message${data != null ? ' $data' : ''}');
  }
}
